torch/lib/c10d/ProcessGroupMPI.hpp

```diff
@@ -111,10 +122,12 @@ class ProcessGroupMPI : public ProcessGroup {
 
     void abort() override;
 
+   virtual std::vector<at::Tensor> result() override;
+
    protected:
     void populateException();
 
-    at::Tensor tensor_;
+    const std::vector<at::Tensor>* outputs_;
```

Comment:
curious, why we previously use a single tensor, but now need a vector of tensors?